zebra-rpc/src/methods/tests/snapshot.rs

```diff
@@ -40,6 +49,105 @@ async fn test_rpc_response_data() {
+            // none of which are compatible with NU5 due to the following consensus rule:
+            //
+            // > [NU5 onward] hashBlockCommitments MUST be set to the value of
+            // > hashBlockCommitments for this block, as specified in [ZIP-244].
+            //
+            // Activating NU5 at a lower height and using the 10 blocks causes a failure in
+            // [`zebra_state::populated_state`].
+            nu5: Some(10),
+            ..Default::default()
+        })
+        .with_network_name("custom_testnet")
+        .to_network();
+
+    // Initiate the snapshots of the RPC responses.
+    let mut settings = insta::Settings::clone_current();
+    settings.set_snapshot_suffix(network_string(&testnet).to_string());
+
+    let blocks: Vec<_> = testnet
+        .blockchain_iter()
+        .map(|(_, block_bytes)| block_bytes.zcash_deserialize_into().unwrap())
+        .collect();
+
+    let (_, state, tip, _) = zebra_state::populated_state(blocks.clone(), &testnet).await;
+
+    let (rpc, _) = RpcImpl::new(
+        "",
+        "",
+        testnet,
+        false,
+        true,
+        Buffer::new(MockService::build().for_unit_tests::<_, _, BoxError>(), 1),
+        state,
+        tip,
+    );
+
+    // Request the treestate by a hash.
+    let rsp = rpc
+        .z_get_treestate(blocks[0].hash().to_string())
+        .await
+        .expect("genesis treestate = no treestate");
+    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_by_hash", rsp));
+
+    // Request the treestate by a hash for a block which is not in the state.
+    let rsp = rpc.z_get_treestate(block::Hash([0; 32]).to_string()).await;
+    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_by_non_existent_hash", rsp));
+
+    // Request the treestate before Sapling activation.
+    let rsp = rpc
+        .z_get_treestate((SAPLING_ACTIVATION_HEIGHT - 1).to_string())
+        .await
+        .expect("no Sapling treestate and no Orchard treestate");
+    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_no_treestate", rsp));
+
+    // Request the treestate at Sapling activation.
+    let rsp = rpc
+        .z_get_treestate(SAPLING_ACTIVATION_HEIGHT.to_string())
+        .await
+        .expect("empty Sapling treestate and no Orchard treestate");
+    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_empty_Sapling_treestate", rsp));
+
+    // Request the treestate for an invalid height.
+    let rsp = rpc
+        .z_get_treestate(EXCESSIVE_BLOCK_HEIGHT.to_string())
+        .await;
+    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_excessive_block_height", rsp));
+
+    // Request the treestate for an unparsable hash or height.
+    let rsp = rpc.z_get_treestate("Do you even shield?".to_string()).await;
+    settings
+        .bind(|| insta::assert_json_snapshot!("z_get_treestate_unparsable_hash_or_height", rsp));
```

Comment:
Optional/Nitpick: It seems the snapshots include the variable name, `tree_state` may be a bit more descriptive?

```suggestion
    let tree_state = rpc
        .z_get_treestate(blocks[0].hash().to_string())
        .await
        .expect("genesis treestate = no treestate");
    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_by_hash", tree_state));

    // Request the treestate by a hash for a block which is not in the state.
    let tree_state = rpc.z_get_treestate(block::Hash([0; 32]).to_string()).await;
    settings
        .bind(|| insta::assert_json_snapshot!("z_get_treestate_by_non_existent_hash", tree_state));

    // Request the treestate before Sapling activation.
    let tree_state = rpc
        .z_get_treestate((SAPLING_ACTIVATION_HEIGHT - 1).to_string())
        .await
        .expect("no Sapling treestate and no Orchard treestate");
    settings.bind(|| insta::assert_json_snapshot!("z_get_treestate_no_treestate", tree_state));

    // Request the treestate at Sapling activation.
    let tree_state = rpc
        .z_get_treestate(SAPLING_ACTIVATION_HEIGHT.to_string())
        .await
        .expect("empty Sapling treestate and no Orchard treestate");
    settings.bind(|| {
        insta::assert_json_snapshot!("z_get_treestate_empty_Sapling_treestate", tree_state)
    });

    // Request the treestate for an invalid height.
    let tree_state = rpc
        .z_get_treestate(EXCESSIVE_BLOCK_HEIGHT.to_string())
        .await;
    settings.bind(|| {
        insta::assert_json_snapshot!("z_get_treestate_excessive_block_height", tree_state)
    });

    // Request the treestate for an unparsable hash or height.
    let tree_state = rpc.z_get_treestate("Do you even shield?".to_string()).await;
    settings.bind(|| {
        insta::assert_json_snapshot!("z_get_treestate_unparsable_hash_or_height", tree_state)
    });
```